### PR TITLE
Restructure Fabric docs (2/n)

### PR DESCRIPTION
**Pull request by lightningforever**
Follow up to #17111 based on feedback by @williamFalcon

### CODE REVIEW

File: docs/source-fabric/api/collectives.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. include:: ../links.rst
+
+####################################
+lightning.fabric.plugins.collectives
+####################################
+
+
+Collectives
```
Comment (carmocca) [inline docs/source-fabric/api/collectives.rst:8]: (pre-existing issue) This is very WIP and experimental. Can you add an "experimental" notice?

```python
    .. warning::  This is an :ref:`experimental <versioning:Experimental API>` feature.
```

File: docs/source-fabric/api/io.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.. include:: ../links.rst
+
+###########################
+lightning.fabric.plugins.io
+###########################
+
+
+IO
```
Comment (carmocca) [inline docs/source-fabric/api/io.rst:8]: Same suggestion

File: docs/source-fabric/index.rst [hunk 3]
```diff
@@ -118,23 +118,34 @@ For alternative ways to install, read the :doc:`installation guide <fundamentals
...
 .. toctree::
-   :maxdepth: 1
-   :caption: Get started in steps
+    :maxdepth: 1
+    :caption: Get started in steps
 
-   Basic <levels/basic>
-   Intermediate <levels/intermediate>
-   Advanced <levels/advanced>
+    Basic <levels/basic>
```
Comment (edenlightning) [inline docs/source-fabric/index.rst:124]:
```suggestion
    Basic skills<levels/basic>
```

File: docs/source-fabric/index.rst [hunk 4]
```diff
@@ -118,23 +118,34 @@ For alternative ways to install, read the :doc:`installation guide <fundamentals
...
+    Intermediate <levels/intermediate>
```
Comment (edenlightning) [inline docs/source-fabric/index.rst:125]:
```suggestion
    Intermediate skills <levels/intermediate>
```

File: docs/source-fabric/index.rst [hunk 5]
```diff
@@ -118,23 +118,34 @@ For alternative ways to install, read the :doc:`installation guide <fundamentals
...
+    Advanced <levels/advanced>
```
Comment (edenlightning) [inline docs/source-fabric/index.rst:126]:
```suggestion
    Advanced skills <levels/advanced>
```

File: docs/source-fabric/index.rst [hunk 6]
```diff
@@ -118,23 +118,34 @@ For alternative ways to install, read the :doc:`installation guide <fundamentals
...
 
 
 .. toctree::
     :maxdepth: 1
-    :name: api
     :caption: Core API Reference
 
     Fabric Arguments <api/fabric_args>
```
[the remark this answers is not in the document]
Comment (lightningforever) [reply docs/source-fabric/index.rst:133]: It has to look like shown in Will's image:

image

File: docs/source-fabric/index.rst [hunk 7]
```diff
@@ -118,23 +118,34 @@ For alternative ways to install, read the :doc:`installation guide <fundamentals
...
     Fabric Methods <api/fabric_methods>
-    Utilities <api/utilities>
-    Full API Reference <api_reference>
+
+
+.. toctree::
+    :maxdepth: 1
+    :caption: Full API Reference
```
[the remark this answers is not in the document]
Comment (lightningforever) [reply docs/source-fabric/index.rst:139]: See my comment above about Will's instructions

**Review by Borda — APPROVED** [review]

**Review by edenlightning — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Pull request merged**